smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/model/SymbolExt.kt

```diff
@@ -42,15 +46,36 @@ fun Symbol.isBoxed(): Boolean {
  * Gets the default value for the symbol if present, else null
  */
 fun Symbol.defaultValue(): String? {
-    // boxed types should always be defaulted to null
-    if (isBoxed()) {
+    val default = getProperty(SymbolProperty.DEFAULT_VALUE_KEY, String::class.java)
+
+    // If shape is boxed (nullable) AND there is no default value set, return nil as default value
+    if (isBoxed() && !default.isPresent) {
         return "nil"
     }
 
-    val default = getProperty(SymbolProperty.DEFAULT_VALUE_KEY, String::class.java)
+    // If default value is present, return default value. Otherwise, return null
     return if (default.isPresent) default.get() else null
 }
 
```

Comment:
The `defaultValueFromClosure` and `defaultValueClosure` below allow saving a closure that takes a `SwiftWriter` and returns `String?` into Symbol properties. This allows handling dependency imports required for a certain default value. For example, a symbol X may have default value `Data("".utf8)`, which needs `Foundation.Data` import for it to compile. `SwiftSymbolProvider` doesn't have access to a `SwiftWriter` instance, so using this closure allows us to delay adding import until the symbol actually gets used by the writer.